students/carlos_novoa/lesson08/circle_class.py

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env python3
+
+import math
+
+"""
+Lesson8 - Circle Class
+"""
+
+
+class Circle(object):
+    version = '0.1'
+
+    def __init__(self, radius=None, diameter=None):
+
+        if radius is None and diameter is None:
+            raise ValueError()
+
+        if radius is not None:
+            try:
+                if radius < 0:
+                    raise ValueError()
+            except ValueError:
+                print('Radius must be positive.')
+                return
+            else:
+                self._radius = radius
+                self._diameter = radius * 2
+                self._area = round(math.pi * radius ** 2.0, 6)
+
+        if diameter is not None:
+            try:
+                if diameter < 0:
+                    raise ValueError()
+            except ValueError:
+                print('Diameter must be positive.')
+                return
+            else:
+                self._radius = diameter / 2
+                self._diameter = diameter
+                self._area = round(math.pi * self._radius ** 2.0, 6)
+
+    @property
+    def diameter(self):
+        return self.radius * 2
+
+    @diameter.setter
+    def diameter(self, diameter=None):
+        self._diameter = diameter
+        self._radius = diameter / 2
+
+    @property
+    def radius(self):
+        return self._radius
+
+    @radius.setter
+    def radius(self, radius=None):
+        self._radius = radius
+        self._diameter = radius * 2
```

Comment:
Don't need to repeat the diameter calculation here, since it will run in the diameter property every time the diameter attribute is accessed. But you also don't need the getter and setter for radius at all per my earlier comment, radius should be defined and set as instance attribute with no extra logic needed (the extra logic lives in the diameter and area getters/setters)